how-it-works/2_chainkeytech/04_chain_key_tokens.subpage.md

```diff
@@ -0,0 +1,66 @@
+1. A chain-key implementation of the signature scheme used for signing transactions on the blockchain hosting the underlying asset must be available, e.g., chain-key ECDSA for Bitcoin. This allows for signing transactions for the chain of the underlying token fully on ICP using threshold cryptography without involving an intermediary.
+
+Those two building blocks comprise the native integration with a particular blockchain. ICP has already been integrated with the Bitcoin network using this approach, an integration with Ethereum is planned. Based on this foundation, chain-key token implementations can be built at the smart contract layer. For a specific underlying token, e.g., Bitcoin, this requires a *chain-key token ledger* smart contract, for which the ICRC-1 token standard should be used. The same open source ledger code base can be used and parameterized for any chain-key token to be deployed. In addition to the chain-key token ledger, a chain-key token smart contract is required. This canister essentially replaces the off-chain intermediary of the traditional wrapping architecture with on-chain functionality for the chain-key token architecture. This canister is also called *minter* as it is responsible for creating (and removing) supply of the chain-key token based on in- and outflowing underlying tokens. It keeps any underlying tokens it receives in on-chain custody as long as corresponding chain-key tokens are in circulation, thus ensuring a 1:1 backing of the chain-key token with the underlying asset. The 1:1 backing of the chain-key token by its underlying token can be verified by anyone by checking that the Bitcoin UTXOs the minter claims to hold indeed are held by Bitcoin addresses controlled by the minter, thereby further enhancing trust.
+
+<figure>
+<img src="/img/how-it-works/chain_key_token_arch.png" alt="Architecture for chain-key tokens" title="Architecture for chain-key tokens" align="center" style="width:1000px">
+<figcaption>High-level architecture for chain-key tokens on ICP</figcaption>
+</figure>
+
+Note that the token ledger and minter functionalities can also be integrated into a single smart contract — a question of software architecture. However, splitting the functionality into two canisters as proposed is a preferable approach for the reason of modularity as the same ledger canister code can be reused for any chain-key token and only the minter needs to be adapted to the mechanics of the token of the underlying chain, particularly to account for differences between UTXO- and account-based underlying chains, like Bitcoin and Ethereum, respectively.
```

Comment:
Okay, you do write that here.   :-)
Maybe it's worthwhile to mention that more canisters may also make sense?

Reply:
Done, and also changed the above text.